src/inference/dev_api/ie_icore.hpp

```diff
@@ -212,19 +213,26 @@ class ICore : public ov::ICore {
 /**
  * @private
  */
-class INFERENCE_ENGINE_API_CLASS(DeviceIDParser) {
-    std::string deviceName;
-    std::string deviceID;
-
+class DeviceIDParser : protected ov::DeviceIDParser {
```

Comment:
it's not used outside of openvino repo, we can remove old usage in current PR and not introduce new instances